File: aogc_customization/aogc/web_form/register_organization/register_organization.py
# import frappe

def get_context(context):
	print(context)
	# do your magic here
	# pass

import frappe
from frappe.utils import random_string
@frappe.whitelist(allow_guest=True)
def create_user_and_supplier():
    user = frappe.get_doc({
        'doctype': 'User',
        'email': "myemail@email.com",
        'first_name': "from code",
        'send_welcome_email': 0,
        'enabled': 0,
        'user_type': 'System User'
    })
    
    # Set password for the user
    user.new_password = random_string(10)
    user.save(ignore_permissions=True)
    user.submit()
    user.db.commit()
    return user
    # Create Supplier
    # supplier = frappe.get_doc({
    #     'doctype': 'Supplier',
    #     'supplier_name': doc.organization_name,
    #     # Add other relevant fields here
    # })
    # supplier.insert()
# Attach the method to the Organization DocType
def setup_hooks():
    frappe.get_doc('Register Organization').before_insert = create_user_and_supplier

File: aogc_customization/aogc/web_form/register_organization/register_organization.js
frappe.ready(function () {

    $('.submit-btn').off("click").on("click", function () {

        var organization_name = frappe.web_form.get_value('organization_name');
        var organization_type = frappe.web_form.get_value('organization_type');
        var tender_service_type = frappe.web_form.get_value('tender_service_type');
        var country = frappe.web_form.get_value('country');
        var organization_address = frappe.web_form.get_value('organization_address');
        var establish_date = frappe.web_form.get_value('establish_date');
        var organization_phone = frappe.web_form.get_value('organization_phone');
        var organization_type = frappe.web_form.get_value('organization_type');
        var organization_name = frappe.web_form.get_value('organization_name');
        var organization_type = frappe.web_form.get_value('organization_type');

        frappe.call({
            type: "POST",
            method: 'aogc_customization.aogc.web_form.register_organization.register_organization.create_user_and_supplier',
            args: {

            },
            callback: function (response) {
                console.log(response);
            }
        });
    });
    // frappe.ui.form.on('Register Organization', {
    //     validate: function(frm) {
    //         alert("working")
    //         // Generate a new password
    //         var new_password = generateRandomPassword();

    //         // Set the password for the user associated with the organization
    //         frappe.call({
    //             method: 'aogc_customization.aogc_customization.aogc.web_form.register_organization.register_organization.py',
    //             args: {
    //                 'user_email': frm.doc.email_address,
    //                 'new_password': new_password
    //             },
    //             callback: function(response) {
    //                 frappe.msgprint('Password set successfully!');
    //             }
    //         });
    //     }
    // });


    function generateRandomPassword() {
        // Implement your own password generation logic here
        // Example implementation generating a random alphanumeric password
        var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
        var password = '';
        for (var i = 0; i < 10; i++) {
            password += characters.charAt(Math.floor(Math.random() * characters.length));
        }
        return password;
    }


})